src/mlpack/methods/ann/layer/radial_basis_function_impl.hpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * @file radial_basis_impl.hpp
+ * @author Himanshu Pathak
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_RBF_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_RBF_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "dropout.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType,
+    typename RegularizerType>
+RBF<InputDataType, OutputDataType, RegularizerType>::RBF() :
+    inSize(0),
+    outSize(0),
+    reset(false)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType,
+    typename RegularizerType>
+RBF<InputDataType, OutputDataType, RegularizerType>::RBF(
+    const size_t inSize,
+    const size_t outSize,
+    arma::mat& centres,
+    RegularizerType regularizer) :
+    inSize(inSize),
+    outSize(outSize),
+    centres(centres),
+    regularizer(regularizer),
+    reset(false)
+{
+  weights.set_size(outSize * inSize + outSize, 1);
+}
+
+template<typename InputDataType, typename OutputDataType,
+    typename RegularizerType>
+void RBF<InputDataType, OutputDataType, RegularizerType>::Reset()
+{
+}
+
+template<typename InputDataType, typename OutputDataType,
+    typename RegularizerType>
+template<typename eT>
+void RBF<InputDataType, OutputDataType, RegularizerType>::Forward(
+    const arma::Mat<eT>& input,
+    arma::Mat<eT>& output)
+{
+  if(!reset)
+  {
+    arma::mat sigmas = arma::mat(1, outSize);
+    sigmas.ones();
+    sigmas = sigmas / outSize;
```

Comment:
Can you remove the sigma parameter for now?
I am referencing this [paper](http://proceedings.mlr.press/v51/que16.pdf). They do not use this parameter.